### PR TITLE
feat(content-switcher): add experimental content-switcher

### PR DESCRIPTION
**Pull request by tw15egan**
Closes https://github.com/IBM/carbon-components/issues/1294

Adds in the experimental `ContentSwitcher`

#### Changelog

**New**

- Experimental Content Switcher

**Changed**

- Added a conditional to check for the existence of text inside button, and renders a span around them if it exists. This is to add a margin on the span, and not the icon so that icon only content switcher renders correctly

**Removed**

- Lots of extra styles

#### Testing / Reviewing

[Staging Link](http://carbon-dev-environment-exp-content-switcher-brave-swan.mybluemix.net/?nav=content-switcher)

screen shot 2018-10-23 at 12 30 22 pm

### CODE REVIEW

**Comment by laurenmrice** [comment]
Everything looks good to me 👍 The spacing of the text within sections should be spaced 16 px from the left instead of centered in the section, but I think that stuff can be tweaked later :)

**Review by alisonjoseph — APPROVED** [review]
LGTM!

File: src/globals/scss/_theme-tokens.scss [hunk 1]
```diff
@@ -191,10 +191,6 @@
   $snippet-background-color: $ui-01 !default !global; // TODO: Define for experimental
   $snippet-border-color: $ui-03 !default !global; // TODO: Define for experimental
 
-  // Content Switcher
-  $content-switcher-border-radius: 4px !default !global;
```
Comment (asudoh) [inline src/globals/scss/_theme-tokens.scss:195]: Is the removal here intentional...?

**Review by asudoh — COMMENTED** [review]
LGTM basically, just one comment - Thanks @tw15egan!

File: src/globals/scss/_theme-tokens.scss (hunk 1 above)
Comment (tw15egan) [reply src/globals/scss/_theme-tokens.scss:195]: Whoops, nope. Meant to switch it to 0

**Review by asudoh — APPROVED** [review]
LGTM 👍 - Thanks @tw15egan for your quick change!

**Comment by asudoh** [comment]
Merging as the remainder of design review effort is only of a tweak. My apologies in case I have a oversight!

**Pull request merged**